File: data/readable_format/cpp/file096.cpp
#include <iostream>
#include <cmath> // For pow()

// Declaration of the square function
double square(double num);

int main() {
    double x = 5.0;
    double result = square(x);

    if (result == 25.0) {
        std::cout << "Test passed: The square of " << x << " is " << result << std::endl;
    } else {
        std::cerr << "Test failed" << std::endl;
    }

    return 0;
}

// Definition of the square function
double square(double num) {
    return std::pow(num, 2);
}

File: data/readable_format/fortran/file096.f90
program squareTest
    implicit none
    real :: x, result

    x = 5.0
    result = square(x)
    if (result == 25.0) then
        print *, "Test passed: The square of", x, "is", result
    else
        print *, "Test failed"
    end if

contains

    function square(num) result(squareResult)
        real, intent(in) :: num
        real :: squareResult
        squareResult = num * num
    end function square

end program squareTest